crates/trans_token/src/storage.rs

```diff
@@ -258,6 +260,65 @@ where
     }
 }
 
+/// Transfer tokens from `sources` to `dests`. Returns an `Err` if any source
+/// has insufficient balance or if the transfer to any destination would
+/// overflow (This can only happen if the total supply doesn't fit in
+/// `token::Amount`).
+pub fn multi_transfer<S>(
+    storage: &mut S,
+    sources: &BTreeMap<(Address, Address), Amount>,
+    dests: &BTreeMap<(Address, Address), Amount>,
+) -> storage::Result<()>
+where
+    S: StorageRead + StorageWrite,
+{
+    // Collect all the accounts whose balance has changed
+    let mut accounts = BTreeSet::new();
+    accounts.extend(sources.keys().cloned());
+    accounts.extend(dests.keys().cloned());
+    // Apply the balance change for each account in turn
+    for ref account @ (ref owner, ref token) in accounts {
+        let unexpected_err = || {
+            storage::Error::new_alloc(
```

Comment:
I think this could be a `new_const` and moved out of the loop

Reply:
> I think this could be a `new_const` and moved out of the loop

Thanks, done.